tests/test_raise_op.py

```diff
@@ -184,3 +187,68 @@ def test_CheckAndRaise_sparse_variable():
     a2 = check_and_raise(aspe1, aspe2.sum() > 2)
     with pytest.raises(ValueError, match="sparse_check"):
         a2.sum().eval()
+
+
+@pytensor.config.change_flags(cxx="")  # For speed-up
+def test_vectorize():
+    floatX = pytensor.config.floatX
+    x = pt.vector("x")
+    y = pt.vector("y")
+    cond = pt.all(y >= 0)
+    out = assert_op(x, cond)
+
+    batch_x = pt.matrix("batch_x", shape=(2, None))
+    batch_y = pt.matrix("batch_y", shape=(2, None))
+
+    test_x = np.arange(3).astype(floatX)
+    test_y = np.arange(4).astype(floatX)
+    test_batch_x = np.arange(6).reshape(2, 3).astype(floatX)
+    test_batch_y = np.arange(8).reshape(2, 4).astype(floatX)
+
+    # Only x is batched
+    vect_out = vectorize_graph(out, {x: batch_x, y: y})
+    assert vect_out.type.shape == (2, None)
+    assert isinstance(vect_out.owner.op, CheckAndRaise)
+    np.testing.assert_array_equal(
+        vect_out.eval({batch_x: test_batch_x, y: test_y}),
+        test_batch_x,
+    )
+    with pytest.raises(AssertionError):
+        vect_out.eval({batch_x: test_batch_x, y: -test_y})
+
+    # Only y is batched
+    vect_out = vectorize_graph(out, {x: x, y: batch_y})
+    assert vect_out.type.shape == (2, None)
+    assert vect_out.owner.op == second  # broadcast
```

Comment:
`y` is never an output of `CheckAndRaise`, so it doesn't need to be vectorized (and it can't, since the Op demands it be a scalar bool). The condition is either a scalar True or False, and broadcasting (and then reducing again with `all`) wouldn't would be a no-op.

Second is the Elemwise implicit form of Broadcasting, that allow easier rewrites during canonicalization. When needed it gets eventually rewritten as an Alloc: https://github.com/pymc-devs/pytensor/blob/9df55cc6c507ea5d78f88166664ab1325d1fd30d/pytensor/tensor/rewriting/basic.py#L1-L23